pkg/services/meta/notifications.go

```diff
@@ -32,14 +32,11 @@ func (m *Meta) subscribeForBlocks(ch chan<- *block.Header) (string, error) {
 	return m.ws.ReceiveHeadersOfAddedBlocks(nil, ch)
 }
 
+// unsubscribeFromBlocks reqauires [Meta.cliM] to be taken.
 func (m *Meta) unsubscribeFromBlocks() {
-	var err error
-	m.cliM.Lock()
-	defer m.cliM.Unlock()
-
 	m.l.Debug("unsubscribing from blocks")
 
-	err = m.ws.Unsubscribe(m.blockSubID)
+	err := m.ws.Unsubscribe(m.blockSubID)
```

Comment:
unrelated to deadlock

Reply:
u refactored var declaration in the fix

Reply:
hm, ok. it looked strange without mutexes, so i changed. reverted then